Assets/HotmaxEditorVR/stateManager.cs

```diff
@@ -40,6 +40,32 @@ public enum rotationModes
     public delegate void objectCollidedWithHandHandler(GameObject value);
     public static event objectCollidedWithHandHandler objectCollidedWithHandEvent;
 
+    public static bool playerIsLocomoting;
+    public delegate void playerIsLocomotingHandler(bool value);
+    public static event playerIsLocomotingHandler playerIsLocomotingEvent;
+
+    public delegate void spawnPageDecrementedHandler();
```

Comment:
so this isn't exactly fitting the stateful paradigm - you're just using this as an event bus. Stateful paradigm would be to explicitly say what spawnPage is currently open. Or whether the whether the menu is open or closed, or what material page is currently open, or what the value of the massScale is.. Mutators need to pass information into the StateManager, and listeners need to hear explicitly what the state is from the StateManager and act accordingly.

Reply:
what do you suggest doing?